searchcore/src/vespa/searchcore/proton/matchengine/matchengine.cpp

```diff
@@ -141,11 +141,11 @@ MatchEngine::performSearch(search::engine::SearchRequest::Source req,
     ret->request = req.release();
     ret->setDistributionKey(_distributionKey);
     if (ret->request->trace().getLevel() > 0) {
-        ret->request->trace().getRoot().setLong("distribution-key", _distributionKey);
+        ret->request->trace().getRoot()->setLong("distribution-key", _distributionKey);
```

Comment:
level > 0 does not imply that root is valid. I strongly encourage lazy creation on access for relevant objects.

Reply:
Fixed as discussed